src/EditorFeatures/CSharp/UseAutoProperty/UseAutoPropertyAnalyzer.cs

```diff
@@ -71,50 +72,53 @@ protected override bool CanExplicitInterfaceImplementationsBeFixed()
 
-                foreach (var typeDeclaration in group)
+                foreach (var typeDeclaration in typeDeclarations)
                 {
                     foreach (var argument in typeDeclaration.DescendantNodesAndSelf().OfType<ArgumentSyntax>())
                     {
-                        AnalyzeArgument(semanticModel, argument, ineligibleFields, cancellationToken);
+                        // An argument will disqualify a field if that field is used in a ref/out position.  
+                        // We can't change such field references to be property references in C#.
+                        if (argument.RefKindKeyword.Kind() != SyntaxKind.None)
+                        {
+                            AddIneligibleFields(semanticModel, argument.Expression, ineligibleFields, cancellationToken);
+                        }
+                    }
+
+                    foreach (var refExpression in typeDeclaration.DescendantNodesAndSelf().OfType<RefExpressionSyntax>())
+                    {
+                        AddIneligibleFields(semanticModel, refExpression.Expression, ineligibleFields, cancellationToken);
                     }
                 }
             }
         }
 
-        protected override ExpressionSyntax GetFieldInitializer(VariableDeclaratorSyntax variable, CancellationToken cancellationToken)
+        protected override ExpressionSyntax GetFieldInitializer(VariableDeclaratorSyntax variable,
+            CancellationToken cancellationToken)
         {
             return variable.Initializer?.Value;
         }
 
-        private void AnalyzeArgument(
-            SemanticModel semanticModel, ArgumentSyntax argument,
+        private static void AddIneligibleFields(
+            SemanticModel semanticModel, ExpressionSyntax expression,
             HashSet<IFieldSymbol> ineligibleFields, CancellationToken cancellationToken)
         {
-            // An argument will disqualify a field if that field is used in a ref/out position.  
-            // We can't change such field references to be property references in C#.
-            if (argument.RefOrOutKeyword.Kind() == SyntaxKind.None)
-            {
-                return;
-            }
-
-            var symbolInfo = semanticModel.GetSymbolInfo(argument.Expression, cancellationToken);
-            AddIneligibleField(symbolInfo.Symbol, ineligibleFields);
+            var symbolInfo = semanticModel.GetSymbolInfo(expression, cancellationToken);
+            AddSymbol(symbolInfo.Symbol);
             foreach (var symbol in symbolInfo.CandidateSymbols)
             {
-                AddIneligibleField(symbol, ineligibleFields);
+                AddSymbol(symbol);
             }
-        }
 
-        private static void AddIneligibleField(ISymbol symbol, HashSet<IFieldSymbol> ineligibleFields)
-        {
-            if (symbol is IFieldSymbol field)
+            void AddSymbol(ISymbol symbol)
```

Comment:
can you still call this "AddIneligibleField"

Reply:
@CyrusNajmabadi I originally did but then realized it might be a little misleading because it could add more than 1 symbol

Reply:
whereas the previous method called 'AddIneligibleField' only took the symbol (I changed that one into a local function)

Reply:
but I'm not saying it's a good name as is...

Reply:
okey dokey

Reply:
@CyrusNajmabadi wait, I didn't look properly that you pointed this at the local function itself... yes I could call this AddIneligibleField, but what about AddIneligibleFields then? did you notice both of them?

Reply:
> should they have names differing in 1 letter only?

That's fine.  It's fairly common in our code base.  A higher level functoin that operates on a collection of items, which calls a helper which works on a single element.

Reply:
> yes I could call this AddIneligibleField

great.